packages/pds/src/services/repo/index.ts

```diff
@@ -229,43 +234,76 @@ export class RepoService {
     await sequencer.sequenceEvt(this.db, seqEvt)
   }
 
+  // rebases are expensive & should be done rarely, we don't try to re-process on concurrent writes
   async rebaseRepo(did: string, swapCommit?: CID) {
-    this.db.assertNotTransaction()
-    const storage = new SqlRepoStorage(this.db, did)
-    const currRoot = await storage.getHead()
-    if (!currRoot) {
-      throw new InvalidRequestError(
-        `${did} is not a registered repo on this server`,
-      )
-    }
-    const repo = await Repo.load(storage, currRoot)
-    const rebaseData = await repo.formatRebase(this.repoSigningKey)
-
-    // rebases are expensive & should be done rarely, we don't try to re-process on concurrent writes
-    await this.serviceTx(async (srvcTx) =>
-      srvcTx.processRebase(did, rebaseData, swapCommit),
-    )
-  }
-
-  async processRebase(
-    did: string,
-    rebaseData: RebaseData,
-    swapCommit?: CID,
-    now?: string,
-  ) {
     this.db.assertTransaction()
-    const storage = new SqlRepoStorage(this.db, did, now)
+    const storage = new SqlRepoStorage(this.db, did, new Date().toISOString())
     const currRoot = await storage.lockHead()
     if (!currRoot) {
       throw new ConcurrentWriteError()
     }
     if (swapCommit && !currRoot.equals(swapCommit)) {
       throw new BadCommitSwapError(currRoot)
     }
-    await Promise.all([
-      storage.applyRebase(rebaseData),
-      this.afterRebaseProcessing(did, rebaseData),
-    ])
+    const rebaseData = await this.formatRebase(did, currRoot)
+    const recordCountBefore = await this.countRecordBlocks(did)
+    await storage.applyRebase(rebaseData)
+    const recordCountAfter = await this.countRecordBlocks(did)
+    // This is purely a dummy check on a very sensitive operation
+    if (recordCountBefore !== recordCountAfter) {
+      throw new Error('Record blocks deleted during rebase. Rolling back')
```

Comment:
I wonder if it may be worth including the DID in this error message to ensure we can track it down.

Reply:
yeah good call